### PR TITLE
DataGrid - Row should not focus on scrolling by pointer (T861577)

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.keyboard_navigation.js [hunk 1]
```diff
@@ -125,9 +126,11 @@ const KeyboardNavigationController = core.ViewController.inherit({
             const clickSelector = `.${ROW_CLASS} > td, .${ROW_CLASS}`;
             const $focusedElement = $(':focus');
             const isFocusedElementCorrect = !$focusedElement.length || $focusedElement.closest($rowsView).length || (browser.msie && $focusedElement.is('body'));
+            const isMobile = devices.current().deviceType !== 'desktop';
+            const pointerEventName = !isMobile ? pointerEvents.down : 'dxclick';
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.keyboard_navigation.js]: import click event from '../events/click' module

Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.keyboard_navigation.js]: Added. Also added unit test.

**Review by vladkovl — DISMISSED** [review]

File: js/ui/grid_core/ui.grid_core.keyboard_navigation.js [hunk 2]
```diff
@@ -721,7 +725,7 @@ const KeyboardNavigationController = core.ViewController.inherit({
     // #endregion Key_Handlers
 
     // #region Click_Handler
-    _pointerDownHandler: function(e) {
+    _clickHandler: function(e) {
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.keyboard_navigation.js]: It is not real click. So, I think that will be better to rename it to _focusClickHandler or something like that

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.keyboard_navigation.js (hunk 2 above)
Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.keyboard_navigation.js]: This handler is not only for focus, if editing enabled e.g.
Suggest rename it to pointerEventHandler

Comment (AntonSermyazhko) [reply js/ui/grid_core/ui.grid_core.keyboard_navigation.js]: Renamed: _clickHandler -> _pointerEventHandler

**Review by jtoming830 — APPROVED** [review]

**Review by Alyar666 — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**